### ISSUE TITLE
Request for training code availability

### ISSUE THREAD

**Issue by AlphabetGitHub** [FrankFeng-23/btfm_project#20]
Hi TESSERA team,

Thank you for releasing TESSERA and making the pre-computed embeddings, inference code, and client libraries openly available. This is an incredibly valuable resource for the community.

I noticed on the official project website (`geotessera.org`) that the **model training code** is listed as part of the open-source ecosystem under the MIT license. I'm very interested in studying the pretraining pipeline described in the CVPR paper (Barlow Twins, global shuffling, mixup, etc.).

However, I couldn't seem to locate the training scripts in this GitHub repository. Could you please point me to the correct location, or let me know if there is a planned timeline for releasing the training code?

Thank you for your time and for your great work on TESSERA!

Best regards,
Renwei Luo

<img width="3750" height="978" alt="Image" src="https://github.com/user-attachments/assets/20262107-6400-4b08-8403-0c1a99e3dede" />

**Comment by FrankFeng-23** [comment]
Thank you for raising the issue. The training code is intentionally kept less visible, as very few users will try to reproduce Tessera’s pretraining given the computational cost. At the same time, we encourage users to rely primarily on the precomputed embeddings rather than interacting directly with the model. However, you can find all the original training code [here](https://github.com/ucam-eo/tessera-util/blob/amd/src/train_multi_gpu_FSDP_QAT.py).
To download the training data, you can simply use the code [here](https://github.com/ucam-eo/tessera/tree/v1.0/tessera_preprocessing). Once the raw data is ready, you’ll need to preprocess it before training using the code [here](https://github.com/ucam-eo/tessera-util/tree/amd/training-data-preprocessing). After that, you can start training.

**Comment by AlphabetGitHub** [comment]
@FrankFeng-23 Thank you for the detailed explanation and for providing the links. I really appreciate it.
